diff --git a/core/lib/Drupal/Core/Database/Statement.php b/core/lib/Drupal/Core/Database/Statement.php
index e7463b3..ae49ab6 100644
--- a/core/lib/Drupal/Core/Database/Statement.php
+++ b/core/lib/Drupal/Core/Database/Statement.php
@@ -51,7 +51,12 @@ public function execute($args = array(), $options = array()) {
       $query_start = microtime(TRUE);
     }
 
-    $return = parent::execute($args);
+    if (is_array($args) && count($args)) {
+      $return = parent::execute($args);
+    }
+    else {
+      $return = parent::execute();
+    }
 
     if (!empty($logger)) {
       $query_end = microtime(TRUE);
